File: app/src/main/java/com/testproject/kaera/ringtestapp/ui/MainActivity.java
package com.testproject.kaera.ringtestapp.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.ViewGroup;

import com.bluelinelabs.conductor.Conductor;
import com.bluelinelabs.conductor.Router;
import com.bluelinelabs.conductor.RouterTransaction;
import com.testproject.kaera.ringtestapp.R;
import com.testproject.kaera.ringtestapp.controllers.TopListController;

import butterknife.BindView;

import static butterknife.ButterKnife.*;

public class MainActivity extends AppCompatActivity implements ActionBarProvider {

    @BindView(R.id.toolbar) Toolbar toolbar;
    @BindView(R.id.controller_container) ViewGroup container;

    private Router router;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bind(this);
        setSupportActionBar(toolbar);
        RingApplication.getComponent().inject(this);
        processRouter(savedInstanceState);
    }

    private void processRouter(Bundle savedInstanceState) {
        router = Conductor.attachRouter(this, container, savedInstanceState);
        if (router.hasRootController()) return;
        router.setRoot(RouterTransaction.with(new TopListController()));
    }

    @Override public void onBackPressed() {
        if (!router.handleBack()) super.onBackPressed();
    }
}

File: app/src/main/java/com/testproject/kaera/ringtestapp/ui/MainActivity.kt
package com.testproject.kaera.ringtestapp.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.bluelinelabs.conductor.Conductor
import com.bluelinelabs.conductor.Router
import com.bluelinelabs.conductor.RouterTransaction
import com.testproject.kaera.ringtestapp.R
import com.testproject.kaera.ringtestapp.controllers.TopListController
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), ActionBarProvider {

    private var router: Router? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)
        processRouter(savedInstanceState)
    }

    private fun processRouter(savedInstanceState: Bundle?) {
        router = Conductor.attachRouter(this, controller_container, savedInstanceState)
        if (router!!.hasRootController()) return
        router!!.setRoot(RouterTransaction.with(TopListController(null)))
    }

    override fun onBackPressed() {
        if (!router!!.handleBack()) super.onBackPressed()
    }
}